lumicks/pylake/kymotracker/detail/binding_times.py

```diff
@@ -0,0 +1,377 @@
+        component_kwargs = {"marker": "o", **component_kwargs}
+        fit_kwargs = {"color": "k", **fit_kwargs}
+
+        components = np.exp(
+            exponential_mixture_log_likelihood_components(
+                self.amplitudes, self.lifetimes, centers, *self.observation_limits
+            )
+        )
+
+        label_maker = lambda a, t: (
+            f"{('' if self.n_components==1 else f'({a:0.2g}) ')}{t:0.2g} sec"
+        )
+
+        plt.hist(self.dwelltimes_sec, bins=bins, density=True, **hist_kwargs)
+        for n in range(self.n_components):
+            label = label_maker(self.amplitudes[n], self.lifetimes[n])
+            plt.plot(centers, components[n], label=label, **component_kwargs)
+
+        label = r"$\ln \mathcal{L} $" + f"= {self.log_likelihood:0.3f}"
+        plt.plot(centers, np.sum(components, axis=0), label=label, **fit_kwargs)
+
+        # rearrange legend entries so that total fit is first
+        legend_components = [[c[-1], *c[:-1]] for c in plt.gca().get_legend_handles_labels()]
+        plt.legend(*legend_components, loc="upper right")
+
+        plt.xscale(spacing)
+        plt.ylabel("Density")
+        plt.xlabel("Dwelltime [s]")
+
+        plt.tight_layout()
+
+
+def exponential_mixture_log_likelihood_components(
+    amplitudes, lifetimes, t, min_observation_time, max_observation_time
+):
+    """Calculate each component of the log likelihood of an exponential mixture distribution.
+
+    The full log likelihood for a single observation is given by:
+        log(L) = log( sum_i( component_i ) )
+
+    with the output of this function being log(component_i) defined as:
+        log(component_i) = log(a_i) - log(N) + log(tau_i) - t/tau_i
+
+    where a_i and tau_i are the amplitude and lifetime of component i and N is a normalization
+    factor that takes into account the minimum and maximum observation times of the experiment.
+
+    Therefore, the full log likelihood is calculated from the output of this function by applying
+    logsumexp(output, axis=0) where the summation is taken over the components.
+
+    Parameters
+    ----------
+    amplitudes : array_like
+        fractional amplitude parameters for each component
+    lifetimes : array_like
+        lifetime parameters for each component in seconds
+    t : array_like
+        dwelltime observations in seconds
+    min_observation_time : float
+        minimum observation time in seconds, generally the line/frame time of the experiment
+    max_observation_time : float
+        maximum observation time in seconds, generally the length of the experiment
+    """
+    amplitudes = amplitudes[:, np.newaxis]
+    lifetimes = lifetimes[:, np.newaxis]
+    t = t[np.newaxis, :]
+
+    norm_factor = np.log(amplitudes) + np.log(
+        np.exp(-min_observation_time / lifetimes) - np.exp(-max_observation_time / lifetimes)
+    )
+    norm_factor = -logsumexp(norm_factor, axis=0)
```

Comment:
I'd prefer the name `minus_log_norm_factor`, since you log-transformed it and took the negative. Personally, I might even prefer to just have this be `log_norm_factor` and then use the minus in the expression below.